lib/srv/db/postgres/test.go

```diff
@@ -39,7 +39,7 @@ import (
 // parameters.
 func MakeTestClient(ctx context.Context, config common.TestClientConfig) (*pgconn.PgConn, error) {
 	// Client will be connecting directly to the multiplexer address.
-	pgconnConfig, err := pgconn.ParseConfig(fmt.Sprintf("postgres://%v", config.Address))
+	pgconnConfig, err := pgconn.ParseConfig(fmt.Sprintf("postgres://%v/?sslmode=verify-full", config.Address))
```

Comment:
Upgrading the pgx library to v4 resulted in a test failure in lib/srv/db.TestProxyRateLimiting/postgres:
https://console.cloud.google.com/cloud-build/builds/ffec62fe-42e1-4fc7-b675-70d993aeff3c?project=ci-account

The library changed how it handled fallback configurations. It creates a non-TLS fallback configuration when sslmode is not set to either require, verify-ca, or verify-full. Previously, the connection loop would return on error and not attempt to connect using a fallback. This was changed in the most recent version and resulted in a different error returned to the test than it expected (connection failure due to non-TLS attempt rather than rate limit exceeded). The fix adds sslmode to our client connection string.

Version 1.8.0 returned when any PgError occurred and didn’t execute a fallback.
https://github.com/jackc/pgconn/blob/3742d6209e5f0a4b70b173477c6c40a0aaf21ce9/pgconn.go#L148-L155
```go
	for _, fc := range fallbackConfigs {
		pgConn, err = connect(ctx, config, fc)
		if err == nil {
			break
		} else if err, ok := err.(*PgError); ok {
			return nil, &connectError{config: config, msg: "server error", err: err}
		}
	}
```

Version 1.11.0 only breaks the loop for 2 types of PgErrors, so fallbacks are executed.
https://github.com/jackc/pgconn/blob/3e5de443149f6d3dbbe1bf7387203c413d0efab4/pgconn.go#L151-L163
```go
	for _, fc := range fallbackConfigs {
		pgConn, err = connect(ctx, config, fc)
		if err == nil {
			break
		} else if pgerr, ok := err.(*PgError); ok {
			err = &connectError{config: config, msg: "server error", err: pgerr}
			ERRCODE_INVALID_PASSWORD := "28P01"                    // worng password
			ERRCODE_INVALID_AUTHORIZATION_SPECIFICATION := "28000" // db does not exist
			if pgerr.Code == ERRCODE_INVALID_PASSWORD || pgerr.Code == ERRCODE_INVALID_AUTHORIZATION_SPECIFICATION {
				break
			}
		}
	}
```

The fix changes the client connection string by setting sslmode to verify-full so a fallback config is not created.
https://github.com/jackc/pgconn/blob/3e5de443149f6d3dbbe1bf7387203c413d0efab4/config.go#L665-L671
```go
	switch sslmode {
	case "allow":
		return []*tls.Config{nil, tlsConfig}, nil
	case "prefer":
		return []*tls.Config{tlsConfig, nil}, nil
	case "require", "verify-ca", "verify-full":
		return []*tls.Config{tlsConfig}, nil
```